src/HealthChecks.AzureKeyVault/AzureKeyVaultHealthCheck.cs

```diff
@@ -11,71 +12,68 @@ namespace HealthChecks.AzureKeyVault
     public class AzureKeyVaultHealthCheck : IHealthCheck
     {
         private readonly AzureKeyVaultOptions _options;
-        public AzureKeyVaultHealthCheck(AzureKeyVaultOptions options)
+        private readonly Uri _keyVaultUri;
+        private readonly TokenCredential _azureCredential;
+
+        public AzureKeyVaultHealthCheck(Uri keyVaultUri, TokenCredential credential, AzureKeyVaultOptions options)
         {
+            _keyVaultUri = keyVaultUri ?? throw new ArgumentNullException(nameof(keyVaultUri));
+            _azureCredential = credential ?? throw new ArgumentNullException(nameof(credential));
             _options = options ?? throw new ArgumentNullException(nameof(options));
         }
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
-                using (var client = CreateClient())
+                foreach (var secret in _options.Secrets)
                 {
-                    foreach (var secret in _options.Secrets)
-                    {
-                        await client.GetSecretAsync(_options.KeyVaultUrlBase, secret, cancellationToken);
-                    }
+                    var secretClient = CreateSecretClient();
+                    await secretClient.GetSecretAsync(_options.KeyVaultUrlBase, secret, cancellationToken);

```

Comment:
The vault URI is no longer passed into the method, but only once into the constructor for all clients. Used this way, you're passing the `KeyVaultUrlBase` as the name of the secret and the `secret` is being passed as the optional version. Same is true for keys and certificates.

Reply:
True, my fault on migrate from older version, I try to fix asap!